exp/runtime/catalog/openapi.go

```diff
@@ -88,9 +95,10 @@ func addHookAndTypesToOpenAPI(openAPI *spec3.OpenAPI, c *Catalog, gvh GroupVersi
 		path = GVHToPath(gvh, "{name}")
 		operation.Parameters = append(operation.Parameters, &spec3.Parameter{
 			ParameterProps: spec3.ParameterProps{
-				Name:     "name",
-				In:       "path",
-				Required: true,
+				Name:        "name",
+				In:          "path",
+				Description: "The handler name. Handlers for the same hook within a single external component implementing Runtime Extensions must have different names",
```

Comment:
Looks like this statement implies that if handlers do not serve the same hook they can share the same name but any two handlers of a single external component irrespective of if they serve the same hook should have different names. If not there will be a collision in the registry.

Reply:
> If not there will be a collision in the registry.

I think the discovery would already fail before we add anything to the registry as we validate that the Names have to be unique there.

Independent of that. I think you're right. Can you open a PR to improve this description?

Reply:
>I think the discovery would already fail before we add anything to the registry as we validate that the Names have to be unique there.

You are right. It would not pass the discovery response validation.

>Independent of that. I think you're right. Can you open a PR to improve this description?

Pr: https://github.com/kubernetes-sigs/cluster-api/pull/6896